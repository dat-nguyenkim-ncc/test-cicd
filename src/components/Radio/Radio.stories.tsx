import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Radio, { RadioProps } from './Radio'
import { Section } from '../primitives'

export default {
  title: 'Form Elements/Radio',
  component: Radio,
} as Meta

const Template: Story<RadioProps> = args => (
  <Section>
    <Radio {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  label: 'Radio',
} as RadioProps
