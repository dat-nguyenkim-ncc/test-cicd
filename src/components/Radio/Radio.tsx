import React from 'react'
import { Box, Flex } from 'theme-ui'
import { Palette } from '../../theme'
import { MapVariantToSize, Size, ViewInterface } from '../../types'
import { Paragraph } from '../primitives'

export type RadioProps = ViewInterface<{
  selected?: boolean
  label: string
  value?: string
  size?: Size
  disabled?: boolean
  onClick?(value: string): void
}>

const sizes: MapVariantToSize = {
  big: 80,
  small: 24,
  tiny: 16,
  normal: 40,
}

const Radio = ({ selected, value, label, sx, size, disabled, onClick }: RadioProps) => {
  const onClickRadio = () => {
    onClick && onClick(value || '')
  }

  return (
    <Flex
      sx={{
        alignItems: 'center',
        cursor: disabled ? undefined : 'pointer',
        opacity: disabled ? 0.5 : 1,
        ...sx,
      }}
      onClick={() => {
        !disabled && onClickRadio()
      }}
    >
      <Box
        sx={{
          borderWidth: 1,
          borderStyle: 'solid',
          borderColor: selected ? 'primary' : 'gray04',
          width: size ? sizes[size] : sizes.small,
          height: size ? sizes[size] : sizes.small,
          minWidth: size ? sizes[size] : sizes.small,
          borderRadius: '100%',
        }}
      >
        {selected && (
          <Flex
            sx={{ height: '100%', width: '100%', justifyContent: 'center', alignItems: 'center' }}
          >
            <Box
              sx={{
                width: size ? +sizes[size] * 0.6 : +sizes.small * 0.6,
                height: size ? +sizes[size] * 0.6 : +sizes.small * 0.6,
                borderRadius: '100%',
                background: Palette.primary,
              }}
            />
          </Flex>
        )}
      </Box>
      <Paragraph sx={{ ml: 2 }}>{label}</Paragraph>
    </Flex>
  )
}

export default Radio
