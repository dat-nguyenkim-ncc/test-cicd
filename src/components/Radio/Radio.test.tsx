import * as React from 'react'
import { shallow } from 'enzyme'
import Radio from './Radio'

describe('Radio', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Radio value={'radio'} selected label="Radio" />)
    expect(wrapper).toMatchSnapshot()
  })
})
