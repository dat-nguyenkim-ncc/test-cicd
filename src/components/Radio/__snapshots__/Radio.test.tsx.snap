// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio matches snapshot 1`] = `ShallowWrapper {}`;
